### ISSUE TITLE
Regression in 2.0.3 preview: NullReferenceException in RelationalDataReader.Dispose() on Migrate()

### ISSUE THREAD

**Issue by OronDF343** [aspnet/EntityFrameworkCore#10147]
The following simple case, which worked in 2.0.0, throws a NullReferenceException in the new preview:

```
var context = services.GetRequiredService<VolunteerDbContext>();
context.Database.Migrate();
```

Note that it happen with any of my DbContexts, not just the one shown above.
The exception is as follows:

```
Object reference not set to an instance of an object.
   at Microsoft.EntityFrameworkCore.Storage.RelationalDataReader.Dispose()
   at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.GetAppliedMigrations()
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.Migrate(DatabaseFacade databaseFacade)
   at TestAspNetCore.Program.Main(String[] args)
```

Additionally, when executing a CountAsync and a ToListAsync the following exceptions are thrown, respectively:

```
Microsoft.EntityFrameworkCore.Query: Error: An exception occurred in the database while iterating the results of a query for context type 'TestAspNetCore.Model.VolunteerDbContext'.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.EntityFrameworkCore.Storage.RelationalDataReader.Dispose()
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncQueryingEnumerable`1.AsyncEnumerator.Dispose()
   at Microsoft.EntityFrameworkCore.Query.AsyncQueryMethodProvider.<GetResult>d__16`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.TaskResultAsyncEnumerable`1.Enumerator.<MoveNext>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.<MoveNext>d__5.MoveNext()

Microsoft.EntityFrameworkCore.Query: Error: An exception occurred in the database while iterating the results of a query for context type 'TestAspNetCore.Model.VolunteerDbContext'.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'WrappedMySqlDataReader'.
   at Microsoft.EntityFrameworkCore.Storage.Internal.WrappedMySqlDataReader.GetReader()
   at Microsoft.EntityFrameworkCore.Storage.Internal.WrappedMySqlDataReader.get_RecordsAffected()
   at Microsoft.EntityFrameworkCore.Storage.RelationalDataReader.Dispose()
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncQueryingEnumerable`1.AsyncEnumerator.Dispose()
   at System.Linq.AsyncEnumerable.SelectEnumerableAsyncIterator`2.Dispose()
   at System.Linq.AsyncEnumerable.AsyncIterator`1.<MoveNext>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.<MoveNext>d__5.MoveNext()

```

### Further technical details
EF Core version: 2.0.1-rtm-203
Database Provider: Pomelo.EntityFrameworkCore.MySql (version 2.0.0 as well as 2.0.1-preview-10072)
Operating system: Windows 7 x64
IDE: Visual Studio 2017 15.4.1

**Comment by ajcvickers** [comment]
@OronDF343 Can you post a project or code listing so we can try to reproduce this?

**Comment by ajcvickers** [comment]
@smitpatel Please investigate this per my email.

**Comment by ajcvickers** [comment]
Filed https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/418 on MySQL.

**Comment by Kagamine** [comment]
Hi @OronDF343, we are fixing this issue, but not finished. For now, you can use .AddDbContext instead of .AddDbContextPool will work.

**Comment by OronDF343** [comment]
Thanks, but unfortunately that workaround did not solve the issue (only one of the DbContext was using .AddDbContextPool and changing that produced the same exceptions). I guess I will downgrade to EF Core 2.0.0 for now (and use non-Async methods for now since 2.0.0 has a bug)

**Comment by ajcvickers** [comment]
@OronDF343 A fix for this issue is being included in the 2.0.1 (2.0.3 meta package) patch.
